Hotfix/ligtcurve updates

thespacedoctor opened the pull request:
This PR addressed issue #345

# CHANGES

- There are 2 plots for unforced and forced. Forced plot does not appear if there is no forced data to plot.
- Each plot has a radio button for log/linear. Flux is plotted in linear scale by default.
- Scale-changing and origin-changing are now synced between top-and-bottom and left-and-right axes. This was a pain in the *&(#!. I have disabled scrolling of the top axis as this was causing problems (and not needed ... use the bottom axis).
- RHS black axis bar now appears.
- Black text for tick markers.
- Double click is now working as expected
- Now using [Rubin's offical LC markers for plots](https://rtn-045.lsst.io/#colorblind-friendly-plots) (Roy already had the correct colours).

RoyWilliams committed:
half way updates

RoyWilliams committed:
fixing lightcurves

RoyWilliams committed:
fixing lightcurves

RoyWilliams reviewed (approved):
Much better than before. Not perfect, but thats life. I have asked Eric for some alerts with forced phot, so we can have a proper look.

RoyWilliams committed:
tweaks to LC

thespacedoctor commented:
LC code is now working as expected. I have moved the legend out of the plot to above the top axis tick-labels. I have also moved the "Standard Photometry" and "Forced Photometry" titles into the legend.

<img width="1359" height="1236" alt="image" src="https://github.com/user-attachments/assets/57f14136-a372-4fc1-9cf3-32353bf72fe3" />

Force photometry duplication has been switched off. Happy to merge?
